src/printSchemaWithDirectives.ts

```diff
@@ -0,0 +1,323 @@
+/** Conveniently represents flow's "Maybe" type https://flow.org/en/docs/types/maybe/ */
+type Maybe<T> = null | undefined | T
+
+import { DirectiveNode, print } from 'graphql'
+import { printBlockString, invariant, inspect } from './graphqlInternal'
+
+import type { GraphQLSchema } from 'graphql'
+import type { GraphQLDirective } from 'graphql'
+import type {
+  GraphQLNamedType,
+  GraphQLArgument,
+  GraphQLInputField,
+  GraphQLScalarType,
+  GraphQLEnumType,
+  GraphQLObjectType,
+  GraphQLInterfaceType,
+  GraphQLUnionType,
+  GraphQLInputObjectType,
+} from 'graphql'
+import { isIntrospectionType } from 'graphql'
+import { isSpecifiedScalarType } from 'graphql'
+import { DEFAULT_DEPRECATION_REASON, isSpecifiedDirective } from 'graphql'
+import {
+  isScalarType,
+  isObjectType,
+  isInterfaceType,
+  isUnionType,
+  isEnumType,
+  isInputObjectType,
+} from 'graphql'
+
+import { astFromValue } from 'graphql'
+
+export function printSchemaWithDirectives(schema: GraphQLSchema): string {
+  return printFilteredSchemaWithDirectives(schema, (n) => !isSpecifiedDirective(n), isDefinedType)
+}
```

Comment:
I don't think we need a custom printer, all graphql versions that nexus support have a schema printer that does print the directives:
https://github.com/graphql/graphql-js/blob/v14.5.8/src/utilities/schemaPrinter.js
https://github.com/graphql/graphql-js/blob/v15.5.1/src/utilities/printSchema.js
And they should be exported by `graphql/utilities`.

Reply:
I believe it only prints the directives declarations, not the directive uses on the SDL types/fields

Reply:
Good point, bit annoying that we have to copy paste/maintain all that code. We should reference the commit it was taken from and maybe the diff vs the original to reduce upgrade burden.

Reply:
Yeah, I'm not too worried about it in the sense that since it's just printing code the paths are pretty simple and easy to validate correctness of (not a lot of business logic that you'd otherwise be worried about with copy/paste).